api/v1/routes/iterations.rb

```diff
@@ -74,13 +74,13 @@ class Iterations < Core
           data['path'] = segments[2..-1].join("/")
         end
 
-        track = Trackler.tracks[data['language']]
+        track = Trackler.tracks[data['language'].downcase]
 
         unless track.exists?
           halt 400, { error: "Unknown language track %s" % data['language'] }.to_json
         end
 
-        unless track.implementations[data['problem'].to_s].exists?
+        unless track.implementations[data['problem'].downcase].exists?
```

Comment:
I have no idea if it would ever be nil, though. My guess is _yes_, because we have the `to_s` there :-) Probably at some point it blew up and I was in a hurry and just slammed a to_s on there without writing a test.